packages/react/src/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,94 @@
+import * as RadixCheckbox from "@radix-ui/react-checkbox";
+import * as RadixLabel from "@radix-ui/react-label";
+import { useId } from "@reach/auto-id";
+import { type ComponentPropsWithRef, type ReactNode, forwardRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import { Text } from "../text";
+import * as styles from "./Checkbox.css";
+import { IconChecked } from "./icons-svg/IconChecked";
+import { IconIndeterminate } from "./icons-svg/IconIndeterminate";
+import { IconUnchecked } from "./icons-svg/IconUnchecked";
+
+type CheckboxProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  {
+    checked?: "indeterminate" | boolean;
+    children: ReactNode;
+    defaultChecked?: "indeterminate" | boolean;
+    disabled?: boolean;
+    endDecorator?: ReactNode;
+    readonly?: boolean;
+  }
+>;
+
+export const Checkbox = forwardRef<HTMLDivElement, CheckboxProps>(
+  (
+    {
+      checked,
+      children,
+      className,
+      defaultChecked,
+      disabled,
+      endDecorator,
+      id: idProp,
+      readonly = false,
+      ...props
+    },
+    ref,
+  ) => {
+    const id = useId(idProp);
+    const isDisabled = Boolean(disabled || readonly);
+
+    return (
+      <Flex
+        ref={ref}
+        {...styles.wrapper(
+          {
+            disabled: isDisabled,
+          },
+          className,
+        )}
+        {...props}
+      >
+        <Flex flexDirection="row" gap="xs">
+          <Box
+            asChild
+            {...styles.indicatorWrapper({
+              disabled: isDisabled,
+            })}
+          >
+            <RadixCheckbox.Root
+              checked={checked}
+              defaultChecked={defaultChecked}
+              disabled={isDisabled}
+              id={id}
+            >
+              <Box asChild {...styles.indicator({ disabled: isDisabled })}>
+                <RadixCheckbox.Indicator>
+                  {IconChecked}
+                  {IconUnchecked}
+                  {IconIndeterminate}
+                </RadixCheckbox.Indicator>
+              </Box>
+            </RadixCheckbox.Root>
+          </Box>
+
+          <RadixLabel.Root htmlFor={id}>
+            <Text {...styles.label({ disabled: disabled })}>{children}</Text>
+          </RadixLabel.Root>
+        </Flex>
+        <RadixLabel.Root htmlFor={id}>
+          <Text {...styles.endDecorator({ disabled: disabled })}>
+            {endDecorator}
+          </Text>
+        </RadixLabel.Root>
```

Comment:
Shouldn't it be handled by the end user? Shouldn't we just append the endDecorator here and leave the styling or wrapping to end user?

Reply:
The endDecorator has a definite style as per the designs.

Reply:
Should it be within RadixLabel or Text , I am not sure. 

c.c. @radnan

Reply:
We're needing a RadixLabel in order to make sure checkbox gets selected when helper text is pressed

Reply:
Yeah I know - not sure if it should be like that with wrapping.

Reply:
we don't need to apply any styles to the decorators - consumers are free to inject whatever they like here

Reply:
We should handle the marginLeft and fontSize at least bhai. No ?

Reply:
Margin left yes, font size no